webapp/test_results_handler.go

```diff
@@ -42,122 +53,106 @@ func testResultsHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	query := r.URL.Query()
-	runSHA, err := shared.ParseSHAParam(r)
+	filter, err := parseTestRunUIFilter(r)
 	if err != nil {
-		http.Error(w, "Invalid query params", http.StatusBadRequest)
-		return
-	}
-
-	var testRunSources []string
-	var testRuns []shared.TestRun
-	if testRunSources, testRuns, err = getTestRunsAndSources(r); err != nil {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
 	data := struct {
-		TestRuns       string
-		TestRunSources string
-		SHA            string
-		Diff           bool
-		Filter         string
-		Labels         string
+		TestRuns      string
+		TestRunFilter testRunsUIFilter
```

Comment:
The field name `TestRunFilter` collides with a commonly used type name in the `shared` package. It might be a bit confusing (especially in the template where it's not obvious which package the field comes from).